cpp/include/raft/spatial/knn/knn.cuh

```diff
@@ -38,65 +38,121 @@ namespace knn {
  *
  * etc...
  *
- * @tparam value_idx
+ * @tparam idx_t
  * @tparam value_t
- * @param inK
- * @param inV
- * @param outK
- * @param outV
+ * @param in_keys
+ * @param in_values
+ * @param out_keys
+ * @param out_values
  * @param n_samples
  * @param n_parts
  * @param k
  * @param stream
  * @param translations
  */
-template <typename value_idx = int64_t, typename value_t = float>
-inline void knn_merge_parts(value_t* inK,
-                            value_idx* inV,
-                            value_t* outK,
-                            value_idx* outV,
+template <typename idx_t = int64_t, typename value_t = float>
+inline void knn_merge_parts(value_t* in_keys,
+                            idx_t* in_values,
+                            value_t* out_keys,
+                            idx_t* out_values,
                             size_t n_samples,
                             int n_parts,
                             int k,
                             cudaStream_t stream,
-                            value_idx* translations)
+                            idx_t* translations)
 {
-  detail::knn_merge_parts(inK, inV, outK, outV, n_samples, n_parts, k, stream, translations);
+  detail::knn_merge_parts(
+    in_keys, in_values, out_keys, out_values, n_samples, n_parts, k, stream, translations);
 }
 
+enum class SelectKAlgo { FAISS, RADIX_8_BITS, RADIX_11_BITS, WARP_SORT };
+
 /**
- * Performs a k-select across column-partitioned index/distance
- * matrices formatted like the following:
- * row1: k0, k1, k2, k0, k1, k2
- * row2: k0, k1, k2, k0, k1, k2
- * row3: k0, k1, k2, k0, k1, k2
+ * Select k smallest or largest key/values from each row in the input data.
  *
- * etc...
+ * If you think of the input data `in_keys` as a row-major matrix with input_len columns and
+ * n_inputs rows, then this function selects k smallest/largest values in each row and fills
+ * in the row-major matrix `out_keys` of size (n_inputs, k).
  *
- * @tparam value_idx
+ * Note, depending on the selected algorithm, the values within rows of `out_keys` are not
```

Comment:
Can you expand here to note the implementations which would not contain sorted values here? It's often assumed the resulting neighbors will be sorted by distance. For example, HDBSCAN and RBC expect that the kth neighbor is the `neighbor at k`, which is used to compute the radius.

Reply:
Added to the description for the enum class.